apps/meteor/app/livechat/server/settingsRegenerator.ts

```diff
@@ -0,0 +1,28 @@
+// Validates settings on DB are correct on structure
+// And deletes invalid ones
+import { Settings } from '@rocket.chat/models';
+
+import { Logger } from '../../../server/lib/logger/Logger';
+
+// Validates settings on DB are correct on structure by matching the ones missing all the required fields
+const logger = new Logger('SettingsRegenerator');
+export async function settingsRegenerator() {
+	const invalidSettings = await Settings.find({
+		// Putting the $and explicit to ensure it's "intentional"
+		$and: [
+			{ value: { $exists: false } },
+			{ type: { $exists: false } },
+			{ public: { $exists: false } },
+			{ packageValue: { $exists: false } },
+			{ blocked: { $exists: false } },
+			{ sorter: { $exists: false } },
+			{ i18nLabel: { $exists: false } },
+		],
+	}).toArray();
+
+	if (invalidSettings.length > 0) {
+		logger.warn(`Found ${invalidSettings.length} invalid settings on DB. Deleting them...`);
+		logger.warn(invalidSettings.map(({ _id }) => _id));
+		await Settings.deleteMany({ _id: { $in: invalidSettings.map(({ _id }) => _id) } });
```

Comment:
a warn with only ids would be weird.. it is better to add them to the same log with a msg:
```suggestion
		const ids = invalidSettings.map(({ _id }) => _id);
		logger.warn({ msg: `Found ${invalidSettings.length} invalid settings on DB. Deleting them...`, ids });
		await Settings.deleteMany({ _id: { $in: ids } });
```